src/Compilers/VisualBasic/Portable/Symbols/Source/SourceMemberMethodSymbol.vb

```diff
@@ -743,6 +743,18 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
                     diagBag)
             End If
 
+            Select Case ContainingType.TypeKind
+                Case TypeKind.Interface, TypeKind.Structure, TypeKind.Enum, TypeKind.Delegate
+                    ' Handles clause is invalid in this context. 
+                    Return Nothing
+
+                Case TypeKind.Class, TypeKind.Module
+                    ' Valid context
+
+                Case Else
+                    Debug.Assert(False)
```

Comment:
If the following code depends on there being a constructor, it seems you should check for one rather than plowing ahead, no?